src/NUnitFramework/tests/Internal/Filters/NotFilterTests.cs

```diff
@@ -67,7 +67,7 @@ public void ExplicitMatchTest()
         {
             var filter = new NotFilter(new CategoryFilter("Dummy"));
 
-            Assert.False(filter.IsExplicitMatch(_topLevelSuite));
+            Assert.True(filter.IsExplicitMatch(_topLevelSuite));
```

Comment:
If you feel like adding a few tests, we could try to correct this. Otherwise, I'd say leave it for another issue, especially since #374 is likely to get us back into this code anyway.